pgmpy/factors/continuous/JointGaussianDistribution.py

```diff
@@ -0,0 +1,337 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+
+import six
+import numpy as np
+from scipy.stats import multivariate_normal
+
+from pgmpy.factors import ContinuousFactor
+
+
+class JointGaussianDistribution(ContinuousFactor):
+    u"""
+    In its most common representation, a multivariate Gaussian distribution
+    over X1...........Xn is characterized by an n-dimensional mean vector μ,
+    and a symmetric n x n covariance matrix Σ.
+    This is the base class for its representation.
+    """
+    def __init__(self, variables, mean, covariance):
+        """
+        Parameters
+        ----------
+        variables: iterable of any hashable python object
+            The variables for which the distribution is defined.
+        mean: n x 1, array like
+            n-dimensional vector where n is the number of variables.
+        covariance: n x n, 2-d array like
+            n x n dimensional matrix where n is the number of variables.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from pgmpy.factors import JointGaussianDistribution as JGD
+        >>> dis = JGD(['x1', 'x2', 'x3'], np.array([[1], [-3], [4]]),
+        ...             np.array([[4, 2, -2], [2, 5, -5], [-2, -5, 8]]))
+        >>> dis.variables
+        ['x1', 'x2', 'x3']
+        >>> dis.mean
+        array([[ 1],
+               [-3],
+               [4]]))
+        >>> dis.covariance
+        array([[4, 2, -2],
+               [2, 5, -5],
+               [-2, -5, 8]])
+        >>> dis.pdf([0,0,0])
+        0.0014805631279234139
+        """
+        no_of_var = len(variables)
+
+        if len(mean) != no_of_var:
+            raise ValueError("Length of mean_vector must be equal to the\
+                                 number of variables.")
+
+        self.mean = np.asarray(np.reshape(mean, (no_of_var, 1)), dtype=float)
```

Comment:
@yashu-seth Is there a particular reason for using `np.asarray()` rather than `np.array()`. `np.array()` creates a new object, thus avoids mutability issues, but same cannot be said about `np.assary()`.